### PR TITLE
Minibatches, dropout, fix Gamma, easier learning algorithm

### PR DESCRIPTION
**Pull request by erikbern**
This changes a bunch of stuff

* Minibatches for faster convergence (although it makes every epoch a lot slower). This also gets rid of the checks for `nan` etc
* Dropout rather than regression. Dropout on the inputs.
* Fit an intercept to the regression model as a separate parameter just so that it's not affected by dropout
* Turns out `tf.igamma` has a bug in the gradient wrt `a` (I'll open an issue with `tensorflow` later). Rather than setting it using gradient descent, do a minor perturbation after each step
* Once gamma was fixed, turns out we can have a much simpler learning rate scheme, not save snapshots etc.

### CODE REVIEW

**Comment by erikbern** [comment]
Will do some more cleanup later
